test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
Well, `find` finds the indices of all *numerically* nonzero values; the only reason why it currently gets non-null indices for strings is that `0 == '\0'` is true. If/when integer-character comparison goes away, it will fail for strings anyway. I'd be fine making `find("string")` a `MethodError`. What do you think, @StefanKarpinski?